### PR TITLE
Introduce release kind 'snapshot'

### PR DESCRIPTION
**Pull request by aleasto**
Not a versioned release per se, but a snapshot of a software being worked on

### CODE REVIEW

**Comment by ximion** [comment]
That's what `development` is though, isn't it? It's a snapshot of the current development release, so either an alpha/beta/rc/etc. ;-)

**Comment by aleasto** [comment]
My use-case is to offer beta/rc releases but not lower grade releases (may be alpha, nightly...) to the user.

More specifically, we'd like to treat Fedora Rawhide as a 'development' release but still offer the next fedora branched release (F38 up until 2 days ago) as an option. Currently that's not possible as both would be marked as 'development'.

**Comment by ximion** [comment]
> My use-case is to offer beta/rc releases but not lower grade releases (may be alpha, nightly...) to the user.

That is the current meaning of `development` though - having AppStream recommend or enforce a specific naming scheme or flow for releases feels like a bad plan, so does changing the meaning of `development`.

> More specifically, we'd like to treat Fedora Rawhide as a 'development' release but still offer the next fedora branched release (F38 up until 2 days ago) as an upgrade option. Currently that's not possible as both would be marked as 'development'.

So, would it make sense to create a new type `snapshot` (or other name) for release types to indicate "this is not an actual release but rather a development snapshot"?
That way, Fedora 38 Alpha would be a development release, while Fedora Rawhide would be perpetually in `snapshot` mode, meaning "this is not an actual release but rather a development snapshot that is being worked on continuously".

**Comment by aleasto** [comment]
Sure, i can do that

**Review by ximion — CHANGES_REQUESTED** [review]

**Comment by ximion** [comment]
Needs a rebase as well ;-)

**Comment by ximion** [comment]
Ping? Would be nice to have that in the next release! ;-)

**Comment by aleasto** [comment]
This change was about making the release information usable for proposing distro upgrades.

We were discussing with @Conan-Kudo and @aleixpol how we would need even more information such as if a release is LTS or not.

I guess the plan was to come up with a full proposal but that kinda faded away... I don't remember the details anymore 😅

**Comment by aleasto** [comment]
Perhaps this should be an attribute of the `development` release?

**Comment by ximion** [comment]
> like the LTS would be an attribute of the `stable` release

Unfortunately "LTS" kind of boils down to being a marketing term with no accepted universal meaning, so we really can't use that term at all in a specification which is all about defining things clearly.
A release can have a `date_eol` property though to define when it is end-of-life.

**Comment by aleasto** [comment]
Right, but if you're on Ubuntu LTS we would want to have the option to offer upgrading to other LTS versions only or to all stable versions

**Comment by Conan-Kudo** [comment]
A generic attribute of some kind that can be used to match based on the value set for it would make more sense. For example, `series="longterm"` would allow filtering based on it.

**Comment by Conan-Kudo** [comment]
No. All we care about is that the value matches for filtering, not what the value actually is.

**Comment by ximion** [comment]
You would still need special logic for whatever the distribution decided to call their long-term support series (maybe it's "leap", maybe it's "lts", ...).
But then again, you will for sure need special logic for most OS upgrades anyway.

A while back I faintly recall the request for adding a `channel` to bundles, which we didn't add because that's part of the bundling system. However, we could add such a tag to a `release`, and I quite like the name `channel` for that (it's a bit less loaded with meaning than "series").

One thing that is annoying is that `<channel>lts</channel>` would be a machine-readable name, and we would possibly also want to add a translatable one, yet we can't have people translate that tag for every new release...
The only way around this that I see would be a release channels registry in the metainfo file, but that would open a pretty big can of worms...

Another alternative would be to introduce the concept of `tags` that already exists for components, but for releases. Or allow a `custom` field for arbitrary data per release.

Would this channel property be needed to be translatable?

File: docs/xml/releases-data.xml [hunk 1]
```diff
@@ -123,6 +123,7 @@
 		<itemizedlist>
 			<listitem><para><literal>stable</literal></para></listitem>
 			<listitem><para><literal>development</literal></para></listitem>
+			<listitem><para><literal>snapshot</literal></para></listitem>
 		</itemizedlist>
 		<para>
 			By default, if no release type is defined, <code>stable</code> is assumed.
```
[the remark this answers is not in the document]
Comment (aleasto) [reply docs/xml/releases-data.xml:129]: I'm not sure why this paragraph wants to instruct other softwares on how to *display* the data... but what should it say for snapshots? Something like don't show this unless manually instructed to?

**Review by ximion — APPROVED** [review]

**Comment by ximion** [comment]
I've added the same tagging mechanism that exists for components to releases as well. So you could now have
```xml
<tags>
  <tag namespace="ubuntu">lts</tags>
</tags>
``` 
on LTS releases, and read that information.

**Comment by Conan-Kudo** [comment]
Cool. 👍🏾

**Pull request merged**